content/en/docs/concepts/instrumenting-library.md

```diff
@@ -0,0 +1,270 @@
+
+```java
+Span span = tracer.spanBuilder("send")
+            .setSpanKind(SpanKind.CLIENT)
+            .startSpan();
+
+// make span active so any nested telemetry is correlated
+// even network calls might have nested layers of spans, logs or events
+try (Scope unused = span.makeCurrent()) {
+  // inject the context
+  propagator.inject(Context.current(), transportLayer, setter);
+  send();
+} catch (Exception e) {
+  span.recordException(e);
+  span.setStatus(StatusCode.ERROR);
+  throw e;
+} finally {
+  span.end();
+}
+```
+
+Here's the full [example of context injection in Java](https://opentelemetry.io/docs/java/manual_instrumentation/#context-propagation).
+
+There might be some exceptions:
+
+- downstream service does not support metadata or prohibits unknown fields
+- downstream service does not define correlation protocols. Is it possible that some future service version will support compatible context propagation? Inject it!
+- downstream service supports custom correlation protocol.
+  - best effort with custom propagator: use OpenTelemetry trace context if compatible.
+  - or generate and stamp custom correlation ids on the span.
+
+### In-process
+
+- **Make your spans active** (aka current): it enables correlating spans with logs and any nested auto-instrumentations.
+- If the library has a notion of context, support **optional** explicit trace context propagation *in addition* to active spans
+  - put spans (trace context) created by library in the context explicitly, document how to access it
+  - allow users to pass trace context in your context
+- Within the library, propagate trace context explicitly - active spans may change during callbacks!
+  - capture active context from users on the public API surface as soon as you can, use it as a parent context for your spans
+  - pass context around and stamp attributes, exceptions, events on explicitly propagated instances
+  - this is essential if you run custom threads, background processing or other things that can break due to async context flow limitations in your language
+
+## Logs
+
+Traces is one of the signals you may emit. Logs and traces complement (not duplicate) each other. Whenever you have something that should have a verbosity, logs are a better choice than traces.
+
+Chances are you library uses logs or some similar mechanism already. If it's a platform logging API or a popular logging library, OpenTelemetry likely has integration already! Integrations usually stamps active trace context on all logs, so users can correlate them.
+
+If you language and ecosystem don't have common logging libraries, use span events for additional details you want to share with users.
+
+## Metrics
+
+[Metrics API](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md) is not stable yet and we don't yet define metrics conventions.
+
+## Misc
+
+### Performance

```

Comment:
maybe get ideas from: https://medium.com/opentelemetry/opentelemetry-beyond-getting-started-5ac43cd0fe26#:~:text=Performance%20best%20practices some of it already covered though